src/components/sidebar/components/sidebar-mobile-controls/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { IconChevronLeft16 } from '@hashicorp/flight-icons/svg-react/chevron-left-16'
+import { IconChevronRight16 } from '@hashicorp/flight-icons/svg-react/chevron-right-16'
+import { useCurrentProduct } from 'contexts'
+import { useSidebarNavData } from 'layouts/sidebar-sidecar/contexts/sidebar-nav-data'
+import Button, { ButtonProps } from 'components/button'
+import s from './sidebar-mobile-controls.module.css'
+
+/**
+ * The base `Button` component used by both `LevelUpButton` and
+ * `LevelDownButton`. Handles setting the `Button` props that are the same for
+ * both buttons.
+ */
+const BaseControlButton = ({
+  className,
+  icon,
+  iconPosition,
+  onClick,
+  text,
+}: ButtonProps) => {
+  return (
+    <Button
+      className={className}
+      color="tertiary"
+      icon={icon}
+      iconPosition={iconPosition}
+      onClick={onClick}
+      text={text}
+    />
+  )
+}
+
+/**
+ * The `Button` that handles moving the user "up" one nav data level in
+ * `Sidebar`.
+ *
+ * Notes:
+ *  - Always has a leading chevron-left icon.
+ *  - Always displays on the lefthand side of the `Sidebar`.
+ */
+const LevelUpButton = ({ onClick, text }: ButtonProps) => {
+  return (
+    <BaseControlButton
+      className={s.levelUpButton}
+      icon={<IconChevronLeft16 />}
+      iconPosition="leading"
+      onClick={onClick}
+      text={text}
+    />
+  )
+}
+/**
+ * The `Button` that handles moving the user "down" one nav data level in
+ * `Sidebar`.
+ *
+ * Notes:
+ *  - Always has a trailing chevron-right icon.
+ *  - If rendered with `LevelUpButton`, displays on the righthand side of the
+ *    `Sidebar`.
+ *  - If rendered alone, displays on the lefthand side of the `Sidebar`.
+ */
+const LevelDownButton = ({ onClick, text }: ButtonProps) => {
+  return (
+    <BaseControlButton
+      className={s.levelDownButton}
+      icon={<IconChevronRight16 />}
+      iconPosition="trailing"
+      onClick={onClick}
+      text={text}
+    />
+  )
+}
+
+/**
+ * Handles rendering either one or both of `LevelUpButton` and
+ * `LevelDownButton`. It determines what to render by consuming data from
+ * `SidebarNavDataContext` using the `useSidebarNavData` hook.
+ */
+const SidebarMobileControls = () => {
```

Comment:
This feels like a really nice level of abstraction with the context, and shows how easy it'll be to continue to compose new functionality with it! 👏